Eatery/Views/Eateries/PopularTimesView.swift

```diff
@@ -0,0 +1,205 @@
+//
+//  PopularTimesView.swift
+//  Eatery
+//
+//  Created by William Ma on 10/10/19.
+//  Copyright © 2019 CUAppDev. All rights reserved.
+//
+
+import SnapKit
+import UIKit
+
+/**
+ Why is this layout delegate needed?
+
+ To animate the expansion of the popular times view, we must change the frame
+ of the scroll view. The popular times view is able to manage its own
+ constraints to change its own height. However, without direct access to the
+ outer scroll view (in CampusEateryMenuViewController), it is unable to
+ cleanly animate the change.
+
+ This delegate is the cleanest (albeit still not super clean) way to access the
+ outer scroll view.
+ */
+protocol PopularTimesViewLayoutDelegate: AnyObject {
+
+    func popularTimesContentSizeDidChange(_ popularTimesView: PopularTimesView)
+
+}
+
+class PopularTimesView: UIView {
+
+    weak var layoutDelegate: PopularTimesViewLayoutDelegate?
+
+    private let eatery: CampusEatery
+
+    private var isExpanded: Bool = true
+    private var showHideButton = UIButton(type: .system)
+
+    private let histogramContainerView = UIView()
+    private var histogramExpandConstraint: Constraint?
+    private var histogramCollapseConstraint: Constraint?
+
+    private var histogramView = HistogramView()
+
+    private let startHour = 6
+    private let overflowedEndHour = 27
+
+    init(eatery: CampusEatery) {
+        self.eatery = eatery
+
+        super.init(frame: .zero)
```

Comment:
can we put `super.init(.zero)` as the first line in the function?

Reply:
The `eatery` property must be initialized before `super.init` is called, so no.